bigquery_etl/cli/udf.py

```diff
@@ -0,0 +1,238 @@
+"""bigquery-etl CLI UDF command."""
+
+import click
+from pathlib import Path
+import pytest
+import re
+import string
+import sys
+import yaml
+
+from ..cli.utils import is_valid_dir, is_authenticated
+from ..format_sql.formatter import reformat
+from ..cli.format import format
+from ..udf import publish_udfs
+from ..docs import validate_docs
+
+UDF_NAME_RE = re.compile(r"^(?P<dataset>[a-zA-z0-9_]+)\.(?P<name>[a-zA-z0-9_]+)$")
+DEFAULT_DEPENDENCY_DIR = "udf_js/lib/"
+DEFAULT_GCS_BUCKET = "moz-fx-data-prod-bigquery-etl"
+DEFAULT_GCS_PATH = ""
+DEFAULT_PROJECT_ID = "moz-fx-data-shared-prod"
+
+
+@click.group()
+@click.pass_context
+def udf(ctx):
+    """Create the CLI group for the UDF command."""
+    ctx.ensure_object(dict)
+    ctx.obj["UDF_DIRS"] = ("udf", "udf_js")
+
+
+@click.group()
+@click.pass_context
+def mozfun(ctx):
+    """Create the CLI group for the mozfun command."""
+    ctx.ensure_object(dict)
+    ctx.obj["UDF_DIRS"] = ("mozfun",)
```

Comment:
CLI commands are essentially the same for mozfun and internal UDFs, except for the default directories. Instead of duplicating every command, the default working directory is passed to the context based on whether the `mozfun` or `udf` command is invoked and then used in the command implementations.